astrokat/observe_main.py

```diff
@@ -376,24 +379,20 @@ def run_observation(opts, kat):
             if opts.all_up and (len(catalogue.filter(el_limit_deg=opts.horizon)) != len(catalogue)):
                 raise NotAllTargetsUpError('Not all targets are currently visible - '
                                            'please re-run the script with --visibility for information')
-        cal_tags = {'Polarisation': 'polcal',
-                    'Flux': 'fluxcal',
-                    'Bandpass': 'bpcal',
-                    'Gain': 'gaincal',
-                    'Delay': 'delaycal',
-                    }
-        target_tags = []
-        for cal_type in cal_tags.keys():
-            target_tags.append('~{}'.format(cal_tags[cal_type]))
-            cal_array = [repr(cal.name) for cal in catalogue.filter(cal_tags[cal_type])]
+        # List sources and their associated functions from observation tags
+        not_cals_filter_list = []
+        for cal_type in cal_tags:
+            not_cals_filter_list.append('~{}'.format(cal_type))
+            cal_array = [repr(cal.name) for cal in catalogue.filter(cal_type)]
             if len(cal_array) < 1:
                 continue  # do not display empty tags
             cal_list = ', '.join(cal_array)
             user_logger.info("{} calibrators are [{}]".format(
```

Comment:
No need to change, but making a CSV string from the `cal_array` and printing inside square brackets is exactly what you get for free just using `cal_array` in the format directly.

Reply:
Please help, I need some advice on string formatting

If I do the CSV string in brackets, I get output:
`2019-08-01 06:55:49Z - GAINCAL ['J0632+1022', 'J1150-0023']`

If I just do
`user_logger.info("{} {}".format(str.upper(cal_type), cal_array))`
I get
`2019-08-01 06:55:49Z - GAINCAL ["'J0632+1022'", "'J1150-0023'"]`

Reply:
Hmmm.  Try using `str` instead of `repr`:  
`cal_array = [str(cal.name) for cal in catalogue.filter(cal_type)]`

Reply:
A, thank you for pointing out the obvious.
I should simply remove the `repr`, the cal.name parameter is a string (duh)